### PR TITLE
HPCC-20035 The remoteread.ecl doesn't compatible with parallel query execution mode

### PR DESCRIPTION
**Pull request by AttilaVamos**
Add WUID dependent suffix for filenames to generate unique logical
filenames pro execution.

Remove logical filenames from dataset, based on the dinamic filenames
can't compare with static key file.

Add cleanup to remove generated logical files.

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:

Tested manually: Execute remoteread.ecl in a loop when parallel query
execution is enabled.

### CODE REVIEW

**Comment by AttilaVamos** [comment]
@richardkchapman Please review. I removed the logical file names from data sets. Can it cause any problem?

**Comment by richardkchapman** [comment]
The problem is that this test is designed to test that the virtual filename is returned properly (among other things).

You could add a unique suffix and remove that in the transform before outputting the result.

**Comment by richardkchapman** [comment]
@AttilaVamos

**Comment by AttilaVamos** [comment]
@richardkchapman Reworked. Unique logical filename trimming ,storing and checking. Please review.

File: testing/regress/ecl/remoteread.ecl [hunk 1]
```diff
@@ -59,9 +61,13 @@ rec_vf_trans := RECORD
  string logicalFile{virtual(logicalfilename)};
 END;
 
-ds := DATASET(fname, rec_vf, FLAT);
-ds_comp := DATASET(fname_comp, rec_vf, FLAT);
-ds_trans := DATASET(fname, rec_vf_trans, FLAT);
+string trimmedLogicalFilename(STRING filename) := FUNCTION
+    return filename[1..(std.str.find(std.str.ToLowerCase(trim(filename)), '-w2')-1)];
```
Comment (richardkchapman) [inline testing/regress/ecl/remoteread.ecl]: This will fail in 1000 years...

**Comment by richardkchapman** [comment]
@AttilaVamos Please squash

**Comment by AttilaVamos** [comment]
@richardkchapman The "This will fail in 1000 years..." problem fixed and commits are squashed.

**Pull request merged**